source/slang/slang-ir-util.cpp

```diff
@@ -230,8 +230,9 @@ bool isSimpleDataType(IRType* type)
         return true;
     case kIROp_ArrayType:
     case kIROp_UnsizedArrayType:
-    case kIROp_PtrType:
         return isSimpleDataType((IRType*)type->getOperand(0));
+    case kIROp_PtrType:
```

Comment:
I am not sure we need to change `isSimpleDataType` here.

The right fix is that if the value type of a pointer type is basic type, then we want to emit OpDebugTypePointer, plus the OpDebugValue logic to update the variable with a pointer typed value.

If the value type isn't basic, then we want to emit the pointer as uint64 (OpDebugTypeBasic), plus the OpDebugValue inst to update the variable with a uint64 value.